### PR TITLE
added get user route

### LINKED ISSUE #5: [BE] Get user/s endpoint
### Tasks

- `GET` request to `/users` which will return data as an array of tutors if we have query parameter `type=tutor`
- `GET` request to `/user/:id` which will return a specific user object

### PR DESCRIPTION
**Pull request by egwilborn**

### CODE REVIEW

**Review by sophiedang0101 — APPROVED** [review]
LGTM

File: package.json
Comment (sirbully) [inline package.json:13]: why did you add these new packages by the way? i think it should work with just the firebase admin dependency 👀

**Review by sirbully — CHANGES_REQUESTED** [review]
I notice there are formatting changes (e.g. double quotes). Before you merge this, I'll make a separate PR first that will address the issue and standardize it for everyone.

File: package.json
Comment (sirbully) [reply package.json:13]: Can you try deleting the new dependencies in this file and just keep the firebase-admin one. Run npm install to cleanup old deps and start the project again to see if it still works without all the extra deps.

Comment (egwilborn) [reply package.json:13]: @sirbully I was getting an error when I downloaded the changes sophie made to connect to firebase. It had said: "Cannot find module '@google-cloud/firestore/build/src/path" so I tried pulling from main again and running npm install and a couple other things but the only thing that worked was running npm install @google-cloud/firestore.
How should I go about making the changes? Do I just make them on my local repo and then recommit?

Comment (sirbully) [reply package.json:13]: That's correct! I pulled your branch on my local and seems to work with just the `firebase-admin` dep. So please remove the duplicate ones in `package.json` and run `npm i` again. Check first if it still works on your end and proceed with the commit 🙇‍♀️

Comment (egwilborn) [reply package.json:13]: I did what you said:
  "dependencies": {
    "firebase-admin": "^11.11.0",
    "next": "13.5.4",
    "react": "^18",
    "react-dom": "^18"
  },
  And I ran npm install
 and I get an error when I try to go to the api/user endpoint:
Error: Cannot find module '@google-cloud/firestore/build/src/path'

Comment (sirbully) [reply package.json:13]: Hey, can you try removing `node-modules` and re-running `npm i`. You can remove `node_modules` by running the following command
```
rm -rvf node_modules
```

Comment (egwilborn) [reply package.json:13]: I did this ^^ and I'm still getting the same error

**Review by sirbully — APPROVED** [review]
LGTM 👍 Feel free to merge when you are ready @egwilborn

File: package.json
Comment (sirbully) [reply package.json:13]: we resolved this during huddle, thanks 🙇‍♀️

**Pull request merged**
